### PR TITLE
SALTO-6765: Fail fetch if credentials are invalid

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Also added throw `Unauthorized Error` when generating the token in OAuth flow (cause it's called before validate credentials)

---

_Additional context for reviewer_
additionally, I wrapped `login` func in try-catch as well, as `authParamsFunc` might make network calls

---
_Release Notes_: 

_Okta adapter_:
- Fix a bug causing fetch not to fail on credentials error

_Zendesk adapter_:
- Fix a bug causing fetch not to fail on credentials error

_Microsoft adapter_:
- Fix a bug causing fetch not to fail on credentials error

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/adapter-components/src/auth/oauth.ts [hunk 1]
```diff
@@ -99,21 +100,30 @@ export const oauthAccessTokenRefresh = async ({
...
+    }
+    return {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  } catch (error) {
+    log.error(
+      'Failed to get access token, error: %s, stack: %s',
+      safeJsonStringify({ data: error?.response?.data, status: error?.response?.status }),
```
Comment (netama) [inline packages/adapter-components/src/auth/oauth.ts:124]: not sure we should log the data here without an option to omit sensitive credential details from the response body, let's start with just the error itself (which it seems we're not currently logging?)

File: packages/microsoft-security-adapter/src/client/connection.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ const getAccessToken = async ({ tenantId, clientId, clientSecret, refreshToken }
       safeJsonStringify({ data: e?.response?.data, status: e?.response?.status }),
       e.stack,
     )
-    throw e
+    throw new clientUtils.UnauthorizedError('Invalid Credentials')
```
Comment (netama) [inline packages/microsoft-security-adapter/src/client/connection.ts:57]: leaving this part to @shoshana-houri

**Review by netama — APPROVED** [review]
thanks! added one comment re the info in the log message

File: packages/adapter-components/src/auth/oauth.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/adapter-components/src/auth/oauth.ts:124]: AFAIK the response data should be safe, but logging the entire error can contain headers sent
by error itself you mean the message ?

File: packages/microsoft-security-adapter/src/client/connection.ts (hunk 2 above)
Comment (shoshana-houri) [reply packages/microsoft-security-adapter/src/client/connection.ts:57]: LGTM :)

File: packages/adapter-components/src/auth/oauth.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/adapter-components/src/auth/oauth.ts]: changed to `message`

**Pull request merged**